From 7e4cb2fe2bbc97da383d0324bdea557422627f16 Mon Sep 17 00:00:00 2001
From: ValdikSS <iam@valdikss.org.ru>
Date: Tue, 20 Jan 2026 07:14:18 +0300
Subject: [PATCH] Add Dual Channel (SBC HD) to Bluetooth Audio Channel Mode
 strings

Add "Dual Channel (SBC HD)" option to the bluetooth_a2dp_codec_channel_mode
string arrays for display in Developer Options.

Adapted for Android 16 QPR1 from LineageOS patches.

Co-authored-by: Flopster101 <nahuelgomez329@gmail.com>
Signed-off-by: Flopster101 <nahuelgomez329@gmail.com>
Change-Id: Ia5df9e8bc2f39c12d4e18c7f31e5a8b52c94d3ef
---
 packages/SettingsLib/res/values-en-rGB/arrays.xml | 2 ++
 packages/SettingsLib/res/values/arrays.xml        | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/packages/SettingsLib/res/values-en-rGB/arrays.xml b/packages/SettingsLib/res/values-en-rGB/arrays.xml
index b53ad98fa245..61171f8b1cfc 100644
--- a/packages/SettingsLib/res/values-en-rGB/arrays.xml
+++ b/packages/SettingsLib/res/values-en-rGB/arrays.xml
@@ -146,11 +146,13 @@
     <item msgid="3014194562841654656">"Use system selection (default)"</item>
     <item msgid="5982952342181788248">"Mono"</item>
     <item msgid="927546067692441494">"Stereo"</item>
+    <item>"Dual Channel (SBC HD)"</item>
   </string-array>
   <string-array name="bluetooth_a2dp_codec_channel_mode_summaries">
     <item msgid="1997302811102880485">"Use system selection (default)"</item>
     <item msgid="8005696114958453588">"Mono"</item>
     <item msgid="1333279807604675720">"Stereo"</item>
+    <item>"Dual Channel (SBC HD)"</item>
   </string-array>
   <string-array name="bluetooth_a2dp_codec_ldac_playback_quality_titles">
     <item msgid="1241278021345116816">"Optimised for Audio Quality (990kbps/909kbps)"</item>
diff --git a/packages/SettingsLib/res/values/arrays.xml b/packages/SettingsLib/res/values/arrays.xml
index 53c9658414c7..5eade50eeb6e 100644
--- a/packages/SettingsLib/res/values/arrays.xml
+++ b/packages/SettingsLib/res/values/arrays.xml
@@ -239,6 +239,7 @@
         <item>Use System Selection (Default)</item>
         <item>Mono</item>
         <item>Stereo</item>
+        <item>Dual Channel (SBC HD)</item>
     </string-array>
 
     <!-- Summaries for Bluetooth Audio Codec Channel Mode selection preference. [CHAR LIMIT=50]-->
@@ -246,6 +247,7 @@
         <item>Use System Selection (Default)</item>
         <item>Mono</item>
         <item>Stereo</item>
+        <item>Dual Channel (SBC HD)</item>
     </string-array>
 
     <!-- Titles for Bluetooth Audio Codec LDAC Playback Quality selection preference. [CHAR LIMIT=70] -->
-- 
2.48.1

